HIPs/hedera-hashgraph.md

```diff
@@ -0,0 +1,302 @@
+
+# Solution
+A `PROJECT` network is based on several individual components.
+The following diagram gives an overview of all components that are needed to run a network and interact with the network.
+
+![Components of a network](../assets/hedera-structure.svg "Components of a network")
+
+The components shown in the diagram will be described in the following sections.
+
+## Consensus Node
+The Consensus Node is the core software that runs a `PROJECT` network like the Hedera public network.
+The node is responsible for maintaining the public ledger, processing requested transactions, holding all accounts and other entities,
+and running the Consensus Service, the Token Service, and the Smart Contract Service to fulfill service requests.
+The node is written in Java and runs on the JVM.
+
+## Mirror Node
+The Mirror Node acts as a query node and stores historical data for the network.
+The node consumes the transaction data provided by Consensus Nodes and may additionally provide value-added services
+such as APIs, auditing, analytics, visibility services, security threat modeling, data monetization services, etc.
+The node can also run additional business logic to support applications.
+
+## Mirror Node Explorer
+A visual Explorer interface built on top of a Mirror node providing a human interface for a network.
+
+## Exchange Rate Tool
+Tool to calculate and send the median of exchange rates from various exchanges to `PROJECT` network to
+establish the cost of transactions.
+
+## JSON-RPC-Relay
+Allows Ethereum clients to interact with a `PROJECT` network.
+This project is an implementation of the Ethereum JSON RPC APIs that delegates to both the
+Consensus and Mirror Nodes to fulfill the JSON RPC Specification.
+
+## Client SDKs (Java, JavaScript, Go, C++, Rust, Swift)
+The SDKs for interacting with a `PROJECT` network: the official distributed consensus platform built using the
+hashgraph consensus algorithm for fast, fair, and secure transactions.
+
+## Solo
+Solo provides a framework and associated tools for conducting end-to-end and full stack tests against a `PROJECT` network.
+
+## Licenses and dependencies of the sub-projects
+All sub-projects of `PROJECT` use the Apache license V2.0 (https://www.apache.org/licenses/LICENSE-2.0.txt).
+Since all the subprojects are part of separate repositories and even created in different programming languages,
+their dependencies differ. Based on that, it doesn’t make sense to provide a global list of dependencies.
+All dependencies and related licenses can be identified by automatic project analysis.
+Today, Snyk is used to create automatic dependency and license overviews of the subprojects.
+The attached file [`dependencies-client-sdks.csv`](../assets/hedera-licenses-sdks.csv) contains a list of all
+dependencies of our SDK projects.
+Next to that, the attached file [`licenses.cvs`](../assets/hedera-licenses.csv) contains an overview of all licenses that
```

Comment:
Seeing a few dependencies that are non-permissive (e.g., GPL) in the list. Is there an intention to replace these dependencies with a permissive license dependency?

Reply:
Can you provide a list of all licences in `hedera-licenses.csv` that are problematic? By doing so we can identify the dependencies and define actions for each of them.

Reply:
This is a list of licenses from hedera-licenses-sdk.csv:

- ""
- "Apache-2.0, MIT"
- "BSD-2-Clause, MIT, Apache-2.0"
- "BSD-3-Clause, GPL-2.0"
- "BSD-3-Clause, MIT"
- "BSD-3-Clause, Unlicense, Unknown"
- "CDDL-1.1, GPL-2.0-with-classpath-exception"
- "GPL-3.0, LGPL-3.0"
- "MIT, CC0-1.0"
- "WTFPL, MIT"
- 0BSD
- Apache-2.0
- BSD-2-Clause
- BSD-3-Clause
- CC-BY-3.0
- CC-BY-4.0
- CC0-1.0
- EDL-1.0
- GPL-2.0
- ISC
- LGPL-2.1
- LGPL-3.0
- MIT
- MPL-2.0
- Public-Domain
- Python-2.0
- Unknown
- Unlicense

The dependencies with "" or Unknown licenses are something that should be researched to determine what the actual license is.

The lines that have "GPL-2.0"  and "CDDL-1.1" are also of concern as these are non-permissive licenses. There may be others, as I am not familiar with all of these licenses.

Also, are those lines that contain multiple licenses "one-of" or something else?

Reply:
This is a list of licenses from hedera-licenses.csv:

- "0BSD, BSD-2-Clause"
- "AFL-2.1, BSD-2-Clause"
- "AFL-2.1, BSD-3-Clause"
- "Apache-2.0, BSD-2-Clause"
- "Apache-2.0, BSD-2-Clause, BSD-3-Clause, BSL-1.0, CC0-1.0, MIT"
- "Apache-2.0, BSD-2-Clause, BSD-3-Clause, EDL-1.0, EPL-2.0, GPL-2.0-with-classpath-exception, MIT, Public-Domain, W3C"
- "Apache-2.0, BSD-2-Clause, MIT, Protobuf"
- "Apache-2.0, BSD-3-Clause"
- "Apache-2.0, BSD-3-Clause, MIT"
- "Apache-2.0, EPL-1.0"
- "Apache-2.0, EPL-2.0, GPL-3.0"
- "Apache-2.0, GPL-2.0, MPL-1.1"
- "Apache-2.0, LGPL-2.1"
- "Apache-2.0, LGPL-2.1, MPL-1.1"
- "Apache-2.0, LGPL-3.0"
- "Apache-2.0, LGPL-3.0-or-later"
- "Apache-2.0, LGPL-3.0-or-later, MIT"
- "Apache-2.0, MIT"
- "Apache-2.0, MPL-1.1"
- "BSD-1-Clause, BSD-2-Clause"
- "BSD-1-Clause, BSD-2-Clause-Views"
- "BSD-2-Clause, BSD-3-Clause"
- "BSD-2-Clause, BSD-3-Clause, BSD-3-Clause-No-Military-License"
- "BSD-2-Clause, BSD-3-Clause, HPND"
- "BSD-2-Clause, GPL-2.0-only"
- "BSD-2-Clause, GPL-3.0"
- "BSD-2-Clause, MIT, Apache-2.0"
- "BSD-3-Clause, Apache-2.0"
- "BSD-3-Clause, GPL-2.0"
- "BSD-3-Clause, LGPL-2.1"
- "BSD-3-Clause, MIT"
- "CC-BY-4.0, MIT"
- "CC0-1.0, GPL-2.0-with-classpath-exception"
- "CDDL-1.1, GPL-2.0-with-classpath-exception"
- "CPL-1.0, EPL-1.0, IPL-1.0"
- "EDL-1.0, EPL-1.0"
- "EDL-1.0, EPL-2.0"
- "EPL-1.0, EDL-1.0"
- "EPL-1.0, GPL-2.0, LGPL-2.1"
- "EPL-1.0, LGPL-2.1"
- "EPL-2.0, GPL-2.0-with-classpath-exception"
- "FTL, GPL-2.0"
- "GPL-2.0, MIT"
- "GPL-2.0-with-classpath-exception, CDDL-1.1"
- "GPL-2.0-with-classpath-exception, MIT"
- "GPL-3.0, LGPL-3.0"
- "GPL-3.0, MIT"
- "MIT, Apache-2.0"
- "MIT, BSD-2-Clause"
- "MIT, BSD-3-Clause"
- "MIT, CC0-1.0"
- "MIT, GPL-2.0"
- "MIT, GPL-3.0-or-later"
- "MIT, Unlicense"
- "MIT, WTFPL"
- "MIT, X11"
- "MIT, Zlib"
- "MPL-2.0, EPL-1.0"
- "Ruby, BSD-2-Clause"
- "Ruby, GPL-2.0"
- "Unlicense, Apache-2.0"
- "WTFPL, MIT"
- AGPL-3.0
- ANTLR-PD
- Apache-2.0
- BSD 0.00
- BSD-2-Clause
- BSD-3-Clause
- BSL-1.0
- BlueOak-1.0.0
- CC-BY-3.0
- CC-BY-3.0-US
- CC-BY-4.0
- CC0-1.0
- CDDL-1.0
- CDDL-1.1
- CPL-1.0
- EDL-1.0
- EPL-1.0
- EPL-2.0
- EUPL-1.1
- GPL-2.0
- GPL-2.0-with-classpath-exception
- GPL-3.0
- ISC
- LGPL-2.1
- LGPL-3.0
- MIT
- MIT-0
- MPL-2.0
- ODC-By-1.0
- OpenSSL
- Public-Domain
- Python-2.0
- Unicode-DFS-2016
- Unknown
- Unlicense
- WTFPL
- ZPL-2.1
- Zlib

Reply:
We should check with LF Legal team on these licenses. cc: @hartm

Reply:
if this conversation goes off GH, please keep me cc'd, I have a bit of background on this topic

Reply:
Tracy, the licenses you reference have modifiers that may be ok with your lawyers, the one I think might be an issue is the CDDL:GPL-2.0-with-classpath-exception, CDDL-1.1,  but as it is dual licensed the exception may make it ok.

Reply:
@tkuhrt thank you for the input.